### ISSUE TITLE
Wrong calculation in derivative with unaryMinus

### ISSUE THREAD

**Issue by gpaucot** [josdejong/mathjs#934]
```
var math = require('mathjs'); // 3.16.2

console.log(math.derivative('1 - 2 * x', 'x').toString());       // => -2 ok
console.log(math.derivative('1 - +2 * x', 'x').toString());      // => -2 ok
console.log(math.derivative('1 - -2 * x', 'x').toString());      // => -2 not ok, should be 2
console.log(math.derivative('1 - (-2) * x', 'x').toString());    // => -2 not ok, should be 2
console.log(math.derivative('1 - (-2 * x)', 'x').toString());    // => -2 not ok, should be 2
console.log(math.derivative('1 - ((-2) * x)', 'x').toString());  // => -2 not ok, should be 2
console.log(math.derivative('-((-2) * x) - 1', 'x').toString()); // => -2 not ok, should be 2
```

BTW awesome work guys thanks ! :+1:

**Comment by josdejong** [comment]
Thanks for reporting this bug @gpaucot

**Comment by josdejong** [comment]
It looks like this issue is originating in `simplify` and not in `derivative`:

```js
math.derivative('1 - -2x', 'x', {simplify: false}).toString(); // Correct: 0 - -2 * 1
math.simplify('0 - -2 * 1').toString()  // Wrong: -2
```

Or simpler:
```js
math.simplify('--2') // Wrong: -2 (should be 2)
```

**Comment by firepick1** [comment]
EEK!  I'm on it. Sorry for the delay--was having fun elsewhere. 😢

**Comment by firepick1** [comment]
Fixed in simplify2 https://github.com/josdejong/mathjs/commit/5b175d56a51923c0efd2ac64e805bc70d3bfa0aa
Added unit test for double unaryMinus. @Jos thanks for triage and diagnosis.

**Comment by gpaucot** [comment]
Thank you very much both ! ;)

**Comment by josdejong** [comment]
Thanks @firepick1 for picking this up, I couldn't immediately pinpoint the underlying cause.

**Comment by josdejong** [comment]
This should be fixed now in `v3.16.3`
